File: Algorithms/JAVA/Sorting/Quick_Sort.java
package Sorting;
import java.util.*;

class Quick_Sort
{
    int size;
    int a[];
    
    public Quick_Sort(int n)
    {
        size=n;
        a=new int[size];
    }
    public void input()
    {
    	Scanner in=new Scanner(System.in);
        System.out.println("Enter the elements of the array:");
        for(int i=0;i<size;i++)
        a[i]=in.nextInt();
        sort(a,0,a.length-1);
    }
    public void sort(int a[],int l,int h)
    {
        if(l<h)
        {
            
            int pi=quick(a,l,h);

            sort(a,l,pi-1);
            sort(a,pi+1,h);
            
        }
    }
    public int quick(int a[],int l,int h)
    {
        int pivot=a[h];
        int i=l-1;
        for(int j=l;j<h;j++){
            if(a[j]<pivot){
                i++;

                int temp=a[i];
                a[i]=a[j];
                a[j]=temp;
            }
        }
        int temp=a[i+1];
        a[i+1]=a[h];
        a[h]=temp;

        return i+1;

    }
    public void display()
    {
        System.out.println("Sorted Array:");
        for(int i:a)
        System.out.println(i+"\t");
    }
    public static void main(String[] args)
    {
        Scanner in=new Scanner(System.in);
        
        System.out.println("Enter the size of the array:");
        int n=in.nextInt();
        
        Quick_Sort ob=new Quick_Sort(n);
        ob.input();
        
        ob.display();
        
    }
}

File: Algorithms/Phython/Sorting/quick_sort.py
#Implementating Quick Sort in Python

#define function for partition and sorting
def Partition(A, p, r):
    x = A[r]
    i = p - 1
    for j in range(p, r):
        if(A[j] <= x):
            i = i + 1
            A[i], A[j] = A[j], A[i]
    A[i + 1], A[r] = A[r], A[i + 1]
    return i + 1

#repeating the process
def QuickSort(A, p, r):
    if(p < r):
        q = Partition(A, p, r)
        QuickSort(A, p, q - 1)
        QuickSort(A, q + 1, r)

#defining a array
a = []

#getting input from user and add them into the array
for i in range(5):
    a.append(eval(input('Enetr: ')))

#calling the function
QuickSort(a, 0, 4)
#print the elements in the sorted array
print(a)